File: app/src/main/java/org/tianjyan/pwd/activity/UpdatePasswordGroupNameDialog.java
package org.tianjyan.pwd.activity;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.WindowManager;
import android.widget.EditText;

import org.tianjyan.pwd.R;
import org.tianjyan.pwd.database.PasswordDBRealm;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class UpdatePasswordGroupNameDialog extends Dialog {

    //region field
    @BindView(R.id.update_passwordGroup_name) EditText editText;
    private PasswordDBRealm mainBinder;
    private String oldGroupName;
    //endregion

    //region onclick
    @OnClick(R.id.update_passwordGroup_cancel)
    public void cancelClick() {
        dismiss();
    }

    @OnClick(R.id.update_passwordGroup_sure)
    public void sureClick() {
        String name = editText.getText().toString().trim();
        if(name.length() > 0){
            if(!name.equals(oldGroupName)){
                mainBinder.updatePasswordGroup(oldGroupName, name);
            }
            dismiss();
        }
    }
    //endregion

    //region function

    //region override
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_update_password_group_name);
        ButterKnife.bind(this);

        editText.setText(oldGroupName);
        editText.requestFocus();
    }
    //endregion

    //region private
    public UpdatePasswordGroupNameDialog(Context context, String oldGroupName, PasswordDBRealm mainBinder){
        super(context, android.R.style.Theme_Translucent_NoTitleBar);
        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE |
                WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
        this.mainBinder = mainBinder;
        this.oldGroupName = oldGroupName;
    }
    //endregion

    //endregion
}

File: app/src/main/kotlin/org/tianjyan/pwd/activity/UpdatePasswordGroupNameDialog.kt
package org.tianjyan.pwd.activity

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import android.view.WindowManager
import android.widget.EditText
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import com.home.young.myPassword.R
import org.tianjyan.pwd.database.PasswordDBRealm

class UpdatePasswordGroupNameDialog(context: Context, private val oldGroupName: String, private val mainBinder: PasswordDBRealm) : Dialog(context, android.R.style.Theme_Translucent_NoTitleBar), Parcelable {
    @BindView(R.id.update_passwordGroup_name) internal var editText: EditText? = null

    constructor(parcel: Parcel) : this(
            TODO("context"),
            parcel.readString(),
            TODO("mainBinder")) {

    }

    @OnClick(R.id.update_passwordGroup_cancel)

    fun cancelClick() {
        dismiss()
    }

    @OnClick(R.id.update_passwordGroup_sure)
    fun sureClick() {
        val name = editText!!.text.toString().trim { it <= ' ' }
        if (name.length > 0) {
            if (name != oldGroupName) {
                mainBinder.updatePasswordGroup(oldGroupName, name)
            }
            dismiss()
        }
    }

    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_update_password_group_name)
        ButterKnife.bind(this)

        editText!!.setText(oldGroupName)
        editText!!.requestFocus()
    }

    init {
        window!!.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE or WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(oldGroupName)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<UpdatePasswordGroupNameDialog> {
        override fun createFromParcel(parcel: Parcel): UpdatePasswordGroupNameDialog {
            return UpdatePasswordGroupNameDialog(parcel)
        }

        override fun newArray(size: Int): Array<UpdatePasswordGroupNameDialog?> {
            return arrayOfNulls(size)
        }
    }
}

